File: components/vehicle-domain/src/main/java/com/vmware/tanzu/data/IoT/vehicles/domains/Vehicle.java
package com.vmware.tanzu.data.IoT.vehicles.domains;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Vehicle {
    private String vin;
    private Long odometer;
    private int  speed;
    private int temperature;
    private boolean checkEngine;
    private GpsLocation gpsLocation;

    public String getId() {
        return vin;
    }

    public void setId(String id) {
        this.vin = id;
    }
}

File: components/IoT-connected-vehicles-domains/src/main/kotlin/com/vmware/tanzu/data/IoT/vehicles/domains/Vehicle.kt
package com.vmware.tanzu.data.IoT.vehicles.domains

import java.time.LocalDateTime

data class Vehicle(
    override var vin: String = "",
    override var odometer: Long = 0,
    override var speed : Int = 0,
    override var temperature: Int = 0,
    override var gpsLocation: GpsLocation? = null,
    var predictions : Set<Prediction>? = null
) : IVehicle {
    var id: String?
        get() = vin;
        set(value)
        {
            if(value !=null)
                vin = value
        };
}